stdlib/LinearAlgebra/src/eigen.jl

```diff
@@ -275,6 +275,13 @@ function eigen(A::AbstractMatrix{T}; permute::Bool=true, scale::Bool=true, sortb
     isdiag(AA) && return eigen(Diagonal(AA); permute=permute, scale=scale, sortby=sortby)
     return eigen!(AA; permute=permute, scale=scale, sortby=sortby, jvl=jvl, jvr=jvr, jce=jce, jcv=jcv)
 end
+function eigen(A::AbstractMatrix{T}; permute::Bool=true, scale::Bool=true, sortby::Union{Function,Nothing}=eigsortby) where T<:Float16
+    AA = copy_oftype(A, eigtype(T))
+    isdiag(AA) && return eigen(Diagonal(AA); permute=permute, scale=scale, sortby=sortby)
+    A = eigen!(AA; permute=permute, scale=scale, sortby=sortby)
+    Eigen(convert(Vector{Float16}, A.values),
+          convert(AbstractArray{Float16}, A.vectors))
```

Comment:
This may fail for complex eigendata, so the conversion must depend on whether `A` is real or complex. Unfortunately, `isreal(A::Eigen)` doesn't work, so we may need to resort to `isreal(A.values)`?

Reply:
That returns bool right? Not sure how we would fit that inside `convert`.

Reply:
I mean you need to have two branches, one in which you convert to `Complex{Float16}` is `isreal(A.values) = false` and the other in which you convert to `Float16` arrays.

Reply:
Oh, gotcha!